testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/miniapptabs/fragments/MiniAppDisplayFragment.kt

```diff
@@ -104,20 +104,30 @@ class MiniAppDisplayFragment : BaseFragment(), PreloadMiniAppWindow.PreloadMiniA
             container,
             false
         )
+        loadMiniApp()
+        return binding.root
+    }
+
+    private fun loadMiniApp() {
+        val activity = requireActivity() as DemoAppMainActivity
         if (isloadNew) {
             isloadNew = false
             initializeMiniAppDisplay(activity)
         } else {
-            if (this::miniAppDisplay.isInitialized)
-                addMiniAppChildView(activity, miniAppDisplay)
-            else
+            if (this::miniAppDisplay.isInitialized) {
+                addMiniAppChildView(
+                    activity,
+                    miniAppView = miniAppIdAndViewMap[Pair(activity.getCurrentSelectedId(), appId)],
+                    miniAppDisplay = miniAppDisplay
+                )
+            } else {
                 initializeMiniAppDisplay(activity)
+            }
         }
-        return binding.root
     }
 
     @Suppress("LongMethod", "ComplexMethod")
-    private fun initializeMiniAppDisplay(activity: Activity) {
+    private fun initializeMiniAppDisplay(activity: DemoAppMainActivity) {
```

Comment:
is that necessary to cast it as `DemoAppMainActivity`? now this fragment can not be work with out `DemoAppMainActivity`, we can not use this fragment anywhere, we might use this fragment in the future so i think it's better not to cast to a specific activity. just use
```
if(activity is DemoAppMainActivity)
   (activity as DemoAppMainActivity).somefunction()

```